### DISCUSSION TITLE
Repository structure

### DISCUSSION THREAD

**Discussion by wormstein** [EdNekebno/DGTCentaurMods#45] [Ideas]
Hey, on my quest to plan a setup automation, I was wondering why our repo structure has a subfolder called DGTCentaurMods. I wish Luca wouls be here to answer why he went this way and if that matters but I think we could rename the repo to DGTCentaurMods and move everything one level higher.

I want to hear your opinion.

**Comment by EdNekebno** [comment]
Yes, I think maybe we give Luca a bit to answer. I quite like the fact that the README, LICENSE, requirements.txt all come outside the main code folder in the way it is now. I'm not really tied to it though.

**Comment by wormstein** [comment]
I like that too. Anyway, if I think is not much of a change in the code to avoid this structure.... Thanks. We'll leave this as it is.

**Comment by lucamonfredo** [comment]
Hey folks!
Sorry for my prolonged absence, but in this period I am in crunch mode with work and unfortunately I am unable to dedicate the time I would like to this project. You are doing a great job and I try to follow you every day, but I can't keep up with you, you are too fast! 😄 
I will try to keep myself up to date and follow along.

Regarding @wormstein question, I was just following [Python's PEP8](https://www.python.org/dev/peps/pep-0008) guidelines, but feel free to modify the structure if you feel the need to. The initial idea was to have a single module with many submodules, that's why I organised things like they are.

**Comment by wormstein** [comment]
Hey Luca, welcome back!
Thanks fot reffering to that PEP8 documentation. It's a good read.

I understand now why you went for dgt_centaur_mods. but it's just soooo long :) And the new one seems not to be PEP8 compliant according to:
> Package and Module Names
Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged.

Any suggestions? 'dgtcm' ot 'dcm' sounds like a choice,

**Comment by wormstein** [comment]
Hey, dont't you think epaper.py should sit in 'display' folder? Would it be hard to switch?

**Reply by EdNekebno** [reply]
Yeh, shouldn't be hard to switch. The idea of pulling it out a bit was that it's separate to the boardfunctions as a feature. Let me finish up what I'm working on first though...I got distracted by a little side feature.

**Comment by wormstein** [comment]
Is .gitignore working for you?

**Comment by wormstein** [comment]
I'll try this:

> If you want to ignore a file that is already checked in, you must untrack the file before you add a rule to ignore it. From your terminal, untrack the file.
`$ git rm --cached FILENAME`

**Reply by EdNekebno** [reply]
Oh...that annoying epaper.jpg :) Maybe I need to do that on mine too.

**Reply by wormstein** [reply]
Just merge my pull request. I have remove it. Make sure when you make other commits you just don't add it. Although .gitignore should keep it away now.

**Reply by wormstein** [reply]
And when you make a new fetch it should be gone in you IDE.

**Reply by EdNekebno** [reply]
OK....merged and pulled. I'll try not to mess it up again :)

**Comment by wormstein** [comment]
Added last night GPL-3.0 License to our repo.

**Reply by EdNekebno** [reply]
Yeh, I saw this. That's great.
